src/app/map-viewer/map-viewer.component.ts

```diff
@@ -0,0 +1,83 @@
+import { OnInit, OnChanges, AfterViewInit, Output, EventEmitter, Input, Component, SimpleChange } from '@angular/core';
+import { LocationEvent, Map, Marker, LatLng } from 'leaflet';
+import { Injectable } from '@angular/core';
+import { MapViewerService } from './map-viewer.service';
+
+import { Subscription } from 'rxjs/Subscription';
+import { User } from '../user/common/user';
+
+@Component({
+  selector: 'my-map-viewer',
+  templateUrl: './map-viewer.component.html',
+  styleUrls: ['./map-viewer.component.scss']
+})
+export class MapViewerComponent implements AfterViewInit {
+  public map: Map;
+  public circles: Array<any>;
+  @Input() mapId: string;
```

Comment:
To start a map, leaflet relies on DOM element Id.
I think, I will modify this component code.